litellm/integrations/literal_ai.py

```diff
@@ -0,0 +1,193 @@
+            prompt_id = None
+            variables = None
+            if "orig_messages" in clean_metadata:
+                orig_messages = clean_metadata.pop("orig_messages")
+                for index, message in enumerate(messages):
+                    orig_message = orig_messages[index]
+                    if literal_prompt := getattr(orig_message, "__literal_prompt__", None):
+                        prompt_id = literal_prompt.get("prompt_id")
+                        variables = literal_prompt.get("variables")
+                        message["uuid"] = literal_prompt.get("uuid")
+                        message["templated"] = True
+            
+            tools = settings.pop("tools", None)
+
+            # only accepts str, int, bool, float for logging
+            for param, value in settings.items():
+                if not isinstance(value, (str, int, bool, float)):
+                    try:
+                        settings[param] = str(value)
+                    except:
+                        # if casting value to str fails don't block logging
+                        pass
+            step: StepDict =  {
+                    "id": clean_metadata.get("step_id", str(uuid.uuid4())),
+                    "name": kwargs.get("model", ""),
+                    "threadId": clean_metadata.get("literalai_thread_id", None),
+                    "parentId": clean_metadata.get("literalai_parent_id", None),
+                    "rootRunId": clean_metadata.get("literalai_root_run_id", None),
+                    "input": None,
+                    "output": None,
+                    "type": "undefined",
+                    "tags": clean_metadata.get("tags", clean_metadata.get("literalai_tags", None)),
+                    "startTime": str(start_time),
+                    "endTime": str(end_time),
+                    "metadata":  clean_metadata,
+                    "generation": {
+                        "promptId": prompt_id,
+                        "variables": variables,
+                        "provider": kwargs.get("custom_llm_provider", "litellm"),
+                        "model": kwargs.get("model", ""),
+                        "duration": (end_time - start_time).total_seconds(),
+                        "settings": settings,
+                        "messages": messages,
+                        "tools": tools,
+                    }
+                }
+            if response_obj is not None and response_obj.get("id", None) is not None:
+                generation_id = litellm.utils.get_logging_id(start_time, response_obj)
+                step["metadata"]["litellm_id"] = generation_id
+                step["generation"]["inputTokenCount"] = response_obj.usage.prompt_tokens
+                step["generation"]["outputTokenCount"] = response_obj.usage.completion_tokens
+                step["generation"]["tokenCount"] = response_obj.usage.prompt_tokens + response_obj.usage.completion_tokens
+
+            if (
```

Comment:
this shouldn't be handled here anymore, as it will require us to update each integration when we add support for different endpoints. 

Use the standard logging payload's `messages` and `response` fields instead - https://github.com/BerriAI/litellm/blob/ab153c3e8dab6be093eba3726ada19da176ee8cc/litellm/types/utils.py#L1331